src/msg/async/EventKqueue.h

```diff
@@ -25,18 +25,33 @@
 
 class KqueueDriver : public EventDriver {
   int kqfd;
-  struct kevent *events;
+  pthread_t mythread;
+  struct kevent *res_events;
   CephContext *cct;
   int size;
 
+  // Keep what we set on the kqfd
+  struct SaveEvent{
+    int fd;
+    int mask;
+  };
+  struct SaveEvent *sav_events;
```

Comment:
@yuyuyu101 
Right, I was hoping not to run into that :), but it does require action.
Would be a lot when 1 OSD has like 5000 fd's open.